tutorials/event_sampler_tutorial.ipynb

```diff
@@ -0,0 +1,582 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "irfs = load_cta_irfs(IRF_FILE)\n",
+    "observation = Observation.create(\n",
+    "    obs_id=1001, pointing=POINTING, livetime=LIVETIME, irfs=irfs\n",
+    ")\n",
+    "\n",
+    "empty = MapDataset.create(WCS_GEOM, energy_axis_true=ENERGY_AXIS_TRUE, migra_axis=MIGRA_AXIS)\n",
+    "maker = MapDatasetMaker(selection=[\"exposure\", \"background\", \"psf\", \"edisp\"])\n",
+    "dataset = maker.run(empty, observation)\n",
+    "\n",
+    "dataset.write('dataset.fits.gz', overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sampling the source and background events"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we can finally add the `Skymodel` we want to simulate to the `Dataset` container:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.models.extend(models_pntpwl)\n",
+    "print(dataset.models)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The next step shows how to sample the events with the `MapdatasetEventSampler` class. The class requests a random number seed generator that we set with `random_state=0`, the `Dataset` and the `Observation` object. From the latter, the `MapDatasetEventSampler` class takes all the meta data information."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sampler = MapDatasetEventSampler(random_state=0)\n",
+    "events = sampler.run(dataset, observation)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The output of the event-sampler is an event list with coordinates, energies and time of arrivals of the source and background events. Source and background events are flagged by the MC_ID identifier (where 0 is the default identifier for the background)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "events.table"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(f\"Source events: {len(np.where(events.table['MC_ID']==2)[0])}\")\n",
+    "print(f\"Background events: {len(np.where(events.table['MC_ID']==0)[0])}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can inspect the properties of the simulated events as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "events.peek()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's write the photon event list to a FITS file:"
```

Comment:
At this stage, we essentially have a simulated observation.
Perhaps link to `analysis_2` tutorial, showing how to use `Datastore.from_events_files()` to use these simulated files?

Reply:
I agree. I've added it.

> Thanks @fabiopintore for this tutorial. If I may suggest a few changes, the new format of having Prerequisites, Context, Objective and Proposed Approach seems to be useful to users. See, eg: https://github.com/gammapy/gammapy/blob/master/tutorials/modeling_2D.ipynb or https://github.com/gammapy/gammapy/blob/master/tutorials/light_curve_flare.ipynb
> 
> It would be nice to reformat the Introduction into this format.
> 
> Also, add this tutorial in https://github.com/gammapy/gammapy/blob/master/tutorials/notebooks.yaml

Thanks @AtreyeeS for the comments. I updated the tutorial following them.